components/core/src/clp/CurlGlobalInstance.cpp

```diff
@@ -0,0 +1,47 @@
+#include "CurlGlobalInstance.hpp"
+
+#include <cstddef>
+#include <mutex>
+
+#include <curl/curl.h>
+
+#include "CurlOperationFailed.hpp"
+#include "ErrorCode.hpp"
+
+namespace clp {
+std::mutex CurlGlobalInstance::m_global_mutex;
+size_t CurlGlobalInstance::m_num_living_instances{0};
+
+CurlGlobalInstance::CurlGlobalInstance() {
+    std::unique_lock<std::mutex> const global_lock{m_global_mutex};
```

Comment:
Looking at the specification https://en.cppreference.com/w/cpp/thread/mutex
I feel [lock_guard](https://en.cppreference.com/w/cpp/thread/lock_guard) is more appropriate since it
`implements a strictly scope-based mutex ownership wrapper`